### ISSUE TITLE
Corrupted Chunks & Crashes

### ISSUE THREAD

**Issue by snibk** [PaperMC/Paper#2455]
"Report this to Paper" error: https://pastebin.com/5qu7ynSh
Crash dump: https://pastebin.com/18WUEqGq

Hi. A few days ago it was brought to my attention that there were a bunch of corrupted chunks in the overworld and nether at seemingly random spots. I first tried to see what caused this, although I ran a world render with WorldBorder and had to roll something back using CoreProtect around the same time this happened so I can't be sure what exactly caused it. I have those two plugins along with Multiverse to manage a separate creative and challenge world, both of which are unaffected by this issue. I noticed that on the Survival world, corrupt chunks can only be found further out, past *roughly* 3600 blocks in any direction from spawn. I tried teleporting to these areas where said chunks are corrupted, and the instant I teleported the server crashed with an error informing me to report it to Paper (see pastebin links).
It's also worth mentioning that when I'd teleport I'd get a huge amount of messages to the console saying "Chunk file ... is in the wrong location, relocating. (Expected ..., got ...)".
I tried running the Minecraft Region Fixer python script and it detected problems, deleted said chunks, but upon running the server again the issues are still there. I even tried deleting the regions themselves from the world folder and they seemed to generate just as they were before, while removing user-created structures. I'm also occasionally getting errors in the console stating, for example: "Invalid chunk: ([52, -228]) Offset: 501 Invalid Size: 5,186>4,096 ./world/region/r.1.-8.mca".

### Plugin list:
BlockLocker, ChestShop, ColoredSigns, ColoredTablist, CoreProtect, DiscordSRV, dynmap, Elevator, EmojiChat, Essentials, EssentialsChat, EssentialsProtect, Giveall, GSit, ImageOnMap, LuckPerms, Multiverse-Core, Multiverse-Inventories, Multiverse-NetherPortals, Multiverse-Portals, NerfPhantoms, OldCombatMechanics, PluginConstructorAPI, PlugMan, PortMinerPlugin, ProtocolLib, RealPing, ServerRestorer, Shopkeepers, SuperVanish, TreeCapitator, Vault, VaultChatFormatter, WorldBorder, WorldEdit, WorldGuard

I have no clue how to reproduce this as of yet. I have a feeling it was caused by an attempt to pregenerate chunks using WorldBorder, but I haven't been able to test that yet and I've refrained from trying to prevent any further damage to the world.

### Paper build number:
git-Paper-166 (MC: 1.14.4) (Implementing API version 1.14.4-R0.1-SNAPSHOT)

I apologize if I'm giving any wrong information or anything, this is my first report here. I'm just not sure where to go from here with what's happening.

**Comment by Spottedleaf** [comment]
Currently ServerRestorer is known to cause some corruption. In their latest version released yesterday, it looks like the biggest offender was resolved however not all seem to be resolved - see https://github.com/ZombieStriker/ServerRestorer/issues/5

I would suggest updating to that and restoring backups for affected regionfiles (and then updating against once the issue I posted is resolved).

Other than we are not sure what else could cause this - and we've seen this for a long time.
